### PR TITLE
RUMM-2961 Record CheckBox implementations in SR

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

In this PR we are adding the capability to record `CheckBox` implementations to our SessionReplay recorder.
Please note that besides the main topic the PR contains also 2 extra patches:
- adding the `MaskAllMapper` versions for the `CheckedTextView` and `CheckBox` implementations as I missed this in the previous PR for `CheckedTextView`
- renaming the `*Mapper` classes to a shorter version for better visibility/readability

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/sessionreplay/RadioCheckBoxesFragment.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+internal class RadioCheckBoxesFragment : Fragment() {
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_radio_checkbox_components, container, false)
+    }
+}
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/sessionreplay/RadioCheckBoxesFragment.kt:25]: Can be even shorter I think.

```suggestion
internal class RadioCheckBoxesFragment : Fragment(R.layout.fragment_radio_checkbox_components)
```

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskAllCheckBoxMapper.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    textWireframeMapper: TextWireframeMapper,
+    stringUtils: StringUtils = StringUtils,
+    uniqueIdentifierGenerator: UniqueIdentifierResolver =
+        UniqueIdentifierResolver,
+    viewUtils: ViewUtils = ViewUtils()
+) : CheckBoxMapper(textWireframeMapper, stringUtils, uniqueIdentifierGenerator, viewUtils) {
+
+    override fun resolveShapeStyle(view: CheckBox, checkBoxColor: String):
+        MobileSegment.ShapeStyle? {
+        return null
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskAllCheckBoxMapper.kt:24]: why do we return `null` here? It is quite not obvious why having `shapeStyle = null` is related to masking all in `CheckBox`. Maybe we should add the comment here?

Same question for the `MaskAllCheckedTextViewMapper `.

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckedTextViewMapper.kt [hunk 3]
```diff
@@ -43,18 +44,29 @@ internal class CheckedTextViewWireframeMapper(
         return stringUtils.formatColorAndAlphaAsHexa(view.currentTextColor, OPAQUE_ALPHA_VALUE)
     }
 
-    override fun resolveCheckBoxSize(view: CheckedTextView, pixelsDensity: Float): Long {
+    override fun resolveCheckBoxBounds(view: CheckedTextView, pixelsDensity: Float): GlobalBounds {
+        val viewGlobalBounds = resolveViewGlobalBounds(view, pixelsDensity)
+        val textViewPaddingRight =
+            view.totalPaddingRight.toLong().densityNormalized(pixelsDensity)
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckedTextViewMapper.kt:50]: Seems not quite RTL compatible? Maybe we should leave the comment / TODO here?

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckBoxMapperTest.kt [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+    private lateinit var mockCheckBox: CheckBox
+
+    @IntForgery(min = 0, max = 0xffffff)
+    var fakeCurrentTextColor: Int = 0
+
+    @FloatForgery(min = 1f, max = 100f)
+    var fakeTextSize: Float = 1f
+
+    @IntForgery(min = 84, max = 100)
```
Comment (0xnm) [inline library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckBoxMapperTest.kt:75]: why 84?

**Review by xgouchet — APPROVED** [review]

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckedTextViewMapper.kt (hunk 3 above)
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckedTextViewMapper.kt:50]: yes is not, we will leave this for later but we have it in mind. That will be another problem to solve at some point as none of our SR content is RTL compatible for now :)

File: library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskAllCheckBoxMapper.kt (hunk 2 above)
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/MaskAllCheckBoxMapper.kt:24]: yes...we return null in order to not fill that checkbox...not showing the selection. I will add a comment.

File: library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckBoxMapperTest.kt (hunk 4 above)
Comment (mariusc83) [reply library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckBoxMapperTest.kt:75]: is the min height I found in my tests but I will add it from the class constant value to be more visible.

**Review by 0xnm — APPROVED** [review]

**Pull request merged**